source/fab/metrics.py

```diff
@@ -171,34 +183,70 @@ def metrics_summary(metrics_folder: Path):
     with open(metrics_folder / JSON_FILENAME, 'rt') as outfile:
         metrics = json.load(outfile)
 
+    logger.info('creating metrics summary')
     logger.debug(f'metrics_summary: got metrics for: {metrics.keys()}')
     metrics_folder.mkdir(parents=True, exist_ok=True)
 
+    #
     # graphs for individual steps
+    #
     step_names = ['preprocess fortran', 'preprocess c', 'compile fortran']
+
+    # histogram
     for step_name in step_names:
+
+        # do we have metrics for this step?
         if step_name not in metrics or step_name not in metrics['steps']:
             continue
 
-        fbase = metrics_folder / step_name.replace(' ', '_')
+        fbase = metrics_folder / ('hist_' + step_name.replace(' ', '_'))
 
         values = metrics[step_name].values()
+        run_times = [value['time_taken'] for value in values]
         total_time = datetime.timedelta(seconds=int(metrics["steps"][step_name]))
 
         if plt:
-            plt.hist(values, 10)
+            plt.hist(run_times, 10)
             plt.suptitle(f'{step_name} histogram\n'
-                         f'{len(values)} files took {total_time}')
+                         f'{len(run_times)} files took {total_time}')
             plt.figtext(0.99, 0.01, f"{metrics['run']['datetime']}", horizontalalignment='right', fontsize='x-small')
             plt.xlabel('time (s)')
             plt.savefig(f"{fbase}.png")
             plt.close()
 
-        top_ten = sorted(metrics[step_name].items(), key=lambda kv: kv[1], reverse=True)[:10]
-        with open(f"{fbase}.txt", "wt") as txt_file:
-            txt_file.write("top ten\n")
-            for i in top_ten:
-                txt_file.write(f"{i}\n")
+    # busby style metrics
+    # https://www.osti.gov/biblio/1393322
+    for step_name in step_names:
+
+        # do we have metrics for this step?
+        if step_name not in metrics or step_name not in metrics['steps']:
+            continue
+
+        fbase = metrics_folder / ('busby_' + step_name.replace(' ', '_'))
+
+        sorted_items = sorted(metrics[step_name].items(), key=lambda item: item[1]['start'])
+        values = [item[1] for item in sorted_items]
+        t0 = values[0]['start']
+        starts = [value['start'] - t0 for value in values]
+        durations = [value['time_taken'] for value in values]
+
+        if plt:
+
+            # larger plot after 500 files
+            size = max(10.0, 10 * len(values) / 500)
+            print('len(values)', len(values))
+            print('size', size)
```

Comment:
Are these left over debug messages? If they are intended to stay should probably make use of the logging framework.

Reply:
yes, thank you!